docs/source/java/vector.rst

```diff
@@ -41,50 +121,56 @@ Note that the current implementation doesn't enforce the rule that Arrow objects
 set/setSafe APIs and concrete subclasses of FieldWriter for populating values.
 
 For example, the code below shows how to build a :class:`BigIntVector`, in this case, we build a
-vector of the range 0 to 7 where the element that should hold the fourth value is nulled::
-
-   BufferAllocator allocator = new RootAllocator(Long.MAX_VALUE);
-
-   BigIntVector vector = new BigIntVector("vector", allocator);
-   vector.allocateNew(8);
-   vector.set(0, 1);
-   vector.set(1, 2);
-   vector.set(2, 3);
-   vector.setNull(3);
-   vector.set(4, 5);
-   vector.set(5, 6);
-   vector.set(6, 7);
-   vector.set(7, 8);
-   vector.setValueCount(8); // this will finalizes the vector by convention.
+vector of the range 0 to 7 where the element that should hold the fourth value is nulled
+
+.. code-block:: Java
+
+    BufferAllocator allocator = new RootAllocator(Long.MAX_VALUE);
```

Comment:
I suggest you cover here or somewhere that most objects in Arrow implement AutoCloseable and in your demos you use try-with-resources to make that obvious. For example, it should be done here with both allocator and vector.

Reply:
Good point. Thanks. 
I have added revised the code snippet, and added a note by the end of the first section.